File: GO/MacPronalts_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestMacPronalts(t *testing.T) {
	tables := []struct {
		prods []int
		qtds  []int
		want  float64
	}{
		{[]int{1001, 1005}, []int{2, 3}, 19.50},
		{[]int{1003}, []int{500}, 1750.00},
		{[]int{1001, 1005, 1003, 1002, 1004}, []int{500, 300, 23, 52, 44}, 2808.50},
	}

	for _, table := range tables {
		got := shopValue(table.prods, table.qtds)

		if got != table.want {
			t.Errorf("Waiting for this %.2f but the return was this: %.2f", table.want, got)
		}
	}
}

File: JS/MacPronalts.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
 const TEST = require('tape')
 const OBJ = require('./MacPronalts.js')
 
 TEST('Starting MacPronalts test...', (t) => {
     //Test 1
     t.assert(OBJ.shopValue([1001, 1005], [2, 3]) === 19.50, "Expect: {1001, 1005}, {2, 3} = 19.50")
     //Test 2
     t.assert(OBJ.shopValue([1003], [500]) === 1750.00, "Expect: {1003}, {500} = 1750.00")
     //Test 3
     t.assert(OBJ.shopValue([1001, 1005, 1003, 1002, 1004], [500, 300, 23, 52, 44]) === 2808.50, "Expect: {1001, 1005, 1003, 1002, 1004}, {500, 300, 23, 52, 44} = 2808.50")
 
     t.end()
 })
 
